plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java

```diff
@@ -0,0 +1,289 @@
+        private static final String TRUSTSTORE_PATH = "truststore.path";
+        private static final String TRUSTSTORE_PASSWORD = "truststore.password";
+        private static final String SSL_PROVIDER = "ssl.provider";
+
+        private final GrpcQueryClient.Config config;
+
+        @Inject
+        public TlsGrpcQueryClientFactory(PinotGrpcServerQueryClientConfig grpcClientConfig, PinotGrpcServerQueryClientTlsConfig tlsConfig)
+        {
+            requireNonNull(grpcClientConfig, "grpcClientConfig is null");
+            requireNonNull(tlsConfig, "tlsConfig is null");
+            ImmutableMap.Builder<String, Object> propertiesBuilder = ImmutableMap.builder();
+            addIfNotNull(propertiesBuilder, CONFIG_MAX_INBOUND_MESSAGE_BYTES_SIZE, String.valueOf(grpcClientConfig.getMaxInboundMessageSize().toBytes()));
+            addIfNotNull(propertiesBuilder, CONFIG_USE_PLAIN_TEXT, String.valueOf(grpcClientConfig.isUsePlainText()));
+            addIfNotNull(propertiesBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_TYPE, tlsConfig.getKeyStoreType());
+            addIfNotNull(propertiesBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_PATH, tlsConfig.getKeyStorePath());
+            addIfNotNull(propertiesBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_PASSWORD, tlsConfig.getKeyStorePassword());
+            addIfNotNull(propertiesBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_TYPE, tlsConfig.getTrustStoreType());
+            addIfNotNull(propertiesBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_PATH, tlsConfig.getTrustStorePath());
+            addIfNotNull(propertiesBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_PASSWORD, tlsConfig.getTrustStorePassword());
+            addIfNotNull(propertiesBuilder, GRPC_TLS_PREFIX + "." + SSL_PROVIDER, tlsConfig.getSslProvider());
+            this.config = new GrpcQueryClient.Config(propertiesBuilder.buildOrThrow());
+        }
+
+        @Override
+        public GrpcQueryClient create(HostAndPort hostAndPort)
+        {
+            return new GrpcQueryClient(hostAndPort.getHost(), hostAndPort.getPort(), config);
+        }
+    }
+
+    public static class PinotGrpcServerQueryClient
+    {
+        private static final CalciteSqlCompiler REQUEST_COMPILER = new CalciteSqlCompiler();
+
+        private final PinotHostMapper pinotHostMapper;
+        private final Map<HostAndPort, GrpcQueryClient> clientCache = new ConcurrentHashMap<>();
+        private final int grpcPort;
+        private final GrpcQueryClientFactory grpcQueryClientFactory;
+        private final Closer closer;
+
+        private PinotGrpcServerQueryClient(PinotHostMapper pinotHostMapper, PinotGrpcServerQueryClientConfig pinotGrpcServerQueryClientConfig, GrpcQueryClientFactory grpcQueryClientFactory, Closer closer)
+        {
+            this.pinotHostMapper = requireNonNull(pinotHostMapper, "pinotHostMapper is null");
+            this.grpcPort = requireNonNull(pinotGrpcServerQueryClientConfig, "pinotGrpcServerQueryClientConfig is null").getGrpcPort();
```

Comment:
Move the null check to a separate line. Inlining it makes it harder to see the core of what this line is doing (i.e., getting the port from the config object).